src/cmdstan-guide/parallelization.Rmd

```diff
@@ -1,37 +1,127 @@
+must be compiled with the `STAN_THREADS` makefile flag. The flag can be 
+supplied in the `make` call as shown below.
+
+```
+make STAN_THREADS=true path/to/model
 ```
-> make STAN_THREADS=TRUE STAN_OPENCL=TRUE /path/to/parallel_logistic
+or on Windows
 ```
+mingw32-make STAN_THREADS=true path/to/model.exe
+```
+The flag can also be written to the `make/local` file.
 
-In addition, for multi-threaded programs, it is necessary to specify the number of available threads
-via the shell environment variable `STAN_NUM_THREADS` which specifies how many threads to run in parallel.
-Generally, this number should not exceed the number of available cores.
-If this variable isn't set, then the program will run single-threaded.
-To run a single chain on a 4-core machine for program `parallel_logistic`:
+### Running
 
+Before running a multi-threaded model, we need to specify the maximum number of threads
+the model can use for a single chain. This is done by setting the environment variable
+`STAN_NUM_THREADS`. On Unix systems we can set it using the `export` command. The 
+following would set the maximum number of threads to use for a chain to four:
 ```
 export STAN_NUM_THREADS=4
-./parallel_logistic sample data file=data.csv ...
 ```
+In Windows PowerShell use the following command:
+```
+Set-Variable -Name STAN_NUM_THREADS -Value "4"
+```
+
+Generally, this number should not exceed the number of available cores.
+If this variable isn't set, then the program will run with only one thread.
+
+Once `STAN_NUM_THREADS` is set, run a model as normal:
+```
+./model sample data file=data.csv ...
+```
+
+## Multi-processing with MPI
+
+In order to use multi-processing with MPI in a Stan model, the models must be 
+rewritten to use the `map_rect` functions. By using MPI, the model can be 
+parallelized across multiple cores or a cluster. MPI with Stan is supported 
+on MacOS and Linux.
+
+### Dependencies
+
+Compiling and running Stan models with MPI requires that the system 
+has an MPI implementation installed. For Unix systems the most commonly used
+implementations are [MPICH](https://www.mpich.org/) and [OpenMPI](https://www.open-mpi.org/).
+
+### Compiling
+
+Once a model is rewritten to use `map_rect`, additional makefile flags 
+must be written to the `make/local`. These are:
+
+- `STAN_MPI`: Enables the use of MPI with Stan if `TRUE`.
+- `CXX`: The name of the MPI C++ compiler wrapper. Typically `"mpicxx"`.
+- `TBB_CXX_TYPE`: The C++ compiler the MPI wrapper wraps. Typically `"gcc"` on Linux and `"clang"` on macOS.
+
+An example of `make/local` on Linux:
+
+```
+STAN_MPI=true
+CXX=mpicxx
+TBB_CXX_TYPE=gcc
+```
+The model is then compiled as normal:
+```
+make path/to/model
+```
+### Running
+
+The Stan model compiled with `STAN_MPI` is then run using an MPI launcher (typically `mpiexec` or `mpirun`).
+The launcher is supplied the path to the built executable and the number of processes to start (`-n`).
+
```

Comment:
MPI Standard suggests `mpiexec`, which takes `-n`. `mpirun` is vendor wrapper, takes `-np` as nproc flag.

Reply:
Thanks!